### PR TITLE
added silent waypoint to iOS

### PR DESCRIPTION
**Pull request by marcotrumpet**
Added silent waypoints to iOS too, following [doc](https://docs.mapbox.com/ios/navigation/guides/turn-by-turn-navigation/route-generation/#silent-waypoints)

### CODE REVIEW

**Comment by eopeter** [comment]
this need to rebase/merge master to only show the impacted changes

**Comment by marcotrumpet** [comment]
> this need to rebase/merge master to only show the impacted changes

I'll suggest to revert your hack fix btw

**Comment by eopeter** [comment]
The hack works and will be left in place until we replace it with something better which will be your PR. It is deployed currently in pub.spec

File: ios/Classes/models/Location.swift [hunk 1]
```diff
@@ -6,11 +6,13 @@ public class Location : Codable
...
     let order: Int?
+    let isSilent: Bool
 
-    init(name: String, latitude: Double?, longitude: Double?, order: Int? = nil) {
+    init(name: String, latitude: Double?, longitude: Double?, order: Int? = nil, isSilent: Bool = false) {
         self.name = name
         self.latitude = latitude
         self.longitude = longitude
         self.order = order
+        self.isSilent = isSilent
```
Comment (eopeter) [inline ios/Classes/models/Location.swift:16]: there is already an `isSilent` on `way_point.dart`. Can we use that instead of having it in 2 places?

Comment (marcotrumpet) [reply ios/Classes/models/Location.swift:16]: that's in swift not in Dart

Comment (eopeter) [reply ios/Classes/models/Location.swift:16]: Ahh. Sorry

**Pull request merged**